runtime/near-vm-runner/src/wasmer2_runner.rs

```diff
@@ -362,22 +362,18 @@ impl Wasmer2VM {
                 }
             };
 
-            let artifact: VMArtifact = match stored_artifact {
-                Some(it) => it,
-                None => {
-                    let executable_or_error = self.compile_and_cache(code, cache)?;
-                    match executable_or_error {
-                        Ok(executable) => self
-                            .engine
-                            .load_universal_executable(&executable)
-                            .map(Arc::new)
-                            .map_err(|err| VMRunnerError::LoadingError(err.to_string()))?,
-                        Err(it) => return Ok(Err(it)),
-                    }
+            Ok(if let Some(it) = stored_artifact {
```

Comment:
I wonder why this `match` statement was replaced with a `if`.

Reply:
Could go either way.  I tend to prefer `if` in those situations because it ends up with one fewer indentation levels.